### ISSUE TITLE
How to configure max_connections  of postgresql?

### ISSUE THREAD

**Issue by samdai** [cloudfoundry-community/postgresql-docker-boshrelease#11]
The default value 100 of max_connections is not enough for my application, I want to change this number bigger, how do I configure it in postgresql-docker-boshrelease?

**Comment by drnic** [comment]
Sadly at the moment the containers do not have a built in way to configure the `postgresql.conf` file. But it would not be hard to implement.

Couple of tid bits of info:
- any parameters passed from CF via `cf create-service postgresql94 free xxx -c '{"MAX_CONNECTIONS": 500}'` are converted into environment variables within the running container (e.g. `$MAX_CONNECTIONS` would be set to `500`)
- you can also arbitrarily set an environment variable for all containers for a specific service plan. For example:

``` yaml
        plans:
          - id: '58dc3912-6e11-11e5-96cc-6c4008a663f0'
            name: 'free'
            container:
              backend: 'docker'
              image: cfcommunity/postgresql
              tag: "9.5"
              persistent_volumes:
                - '/data'
              environment:
              - 'MAX_CONNECTIONS=500'
```

(see https://github.com/cloudfoundry-community/cf-containers-broker/blob/master/app/models/docker_manager.rb#L327-L333)

Finally, you could test for an environment variable and then do a search-replace into `postgresql.conf` at start time, like we do here: https://github.com/cloudfoundry-community/postgresql-docker-boshrelease/blob/master/images/postgresql94/scripts/run.sh#L12

E.g. perhaps try adding something like (written without testing):

```
if [[ "${MAX_CONNECTIONS}X" != "X" ]]; then
  sed -i -e"s/^.*max_connections =.*$/max_connections = $MAX_CONNECTIONS/" $DATA_DIR/postgresql.conf
fi
```

**Comment by drnic** [comment]
Perhaps lower case env variables look nicer; and match the formatting of the `postgresql.conf` file

**Comment by samdai** [comment]
Hi drnic,  I have tried your method, but didn't work. I have a question here, after I add your script to https://github.com/cloudfoundry-community/postgresql-docker-boshrelease/blob/master/images/postgresql94/scripts/run.sh, do I need create a new release to include this change? if need, how do I create a new release?  or just run 

```
templates/make_manifest warden broker embedded
bosh -n deploy
```

**Comment by samdai** [comment]
I think your method should work, but need create a new release to include the code change in run.sh, am I right?

**Comment by drnic** [comment]
You could test it by "docker build" the image yourself, push to your own docker registry account and deploy that.

**Comment by drnic** [comment]
I thought I might have documented this in the images/ folder. Sorry if I didn't!

**Comment by samdai** [comment]
Hello @drnic ,
I have tested it by "docker build" the image, then run the generated image in docker using docker toolbox, the setting max_connections is updated. But I don't know how to convert this docker image into bosh release.
I tried the following steps:
(1) Upload the generate docker image to http://downloads.xxx.com/docker/
(2)Update the content of ~/postgresql-docker-boshrelease/packages/cfcommunity_postgresql_9_3_image/packaging as below

```
set -e; set -u
cd $BOSH_INSTALL_TARGET
wget http://downloads.xxx.com/docker/cfcommunity_postgresql_9_3_image.tgz
```

(3)Then execute the following command, but found the updated image is not included in the release, still use the previous release,   the value of setting max_connections is not updated. Do I miss something?

```
cd postgresql-docker-boshrelease

bosh create release --force
bosh -n upload release
templates/make_manifest warden broker embedded
bosh -n deploy
```
